index.d.ts

```diff
@@ -1,31 +1,64 @@
-import { FastifyPlugin, FastifyReply, FastifyRequest } from 'fastify';
-import { Input, InputObject } from 'into-stream';
-import { Stream } from 'stream';
-import { BrotliOptions, ZlibOptions } from 'zlib';
+import * as fastify from 'fastify'
+import { FastifyPluginCallback, FastifyReply, FastifyRequest, RawServerBase, RawServerDefault } from 'fastify'
+import { Input, InputObject } from 'into-stream'
+import { Stream } from 'stream'
+import { BrotliOptions, ZlibOptions } from 'zlib'
+
+type EncodingToken = 'br' | 'deflate' | 'gzip' | 'identity';
+
+export interface FastifyCompressOptions {
+  brotliOptions?: BrotliOptions;
+  customTypes?: RegExp;
+  encodings?: EncodingToken[];
+  forceRequestEncoding?: EncodingToken;
+  global?: boolean;
+  inflateIfDeflated?: boolean;
+  onInvalidRequestPayload?: (encoding: string, request: FastifyRequest, error: Error) => Error | undefined | null;
+  onUnsupportedEncoding?: (encoding: string, request: FastifyRequest, reply: FastifyReply) => string | Buffer | Stream;
+  onUnsupportedRequestEncoding?: (encoding: string, request: FastifyRequest, reply: FastifyReply) => Error | undefined | null;
+  requestEncodings?: EncodingToken[];
+  threshold?: number;
+  zlib?: unknown;
```

Comment:
Previously it was defined as `NodeModule`, but I don't think it's ok, example :
```js
const zlib = require('zlib')

app.register(fastifyCompress, {
  zlib: zlib // this will trigger a typescript argument error if set to NodeModule
})
```
`unknown` is more suited here I think. WDYT ?